### ISSUE TITLE
test_batchnorm has confusing use of data_format

### ISSUE THREAD

**Issue by mikesol** [keras-team/keras#9144]
In `backend_test.py::TestBackend::test_batchnorm`, there is a for loop to loop over two different data formats.  However, the data format is never used in the batch normalization test, so it is not clear why the format is there.  I would recommend either adding a comment about why the data_format needs to be part of the test or just remove it.  Thanks!

**Comment by anj-s** [comment]
The test is being run on two different data_format values. The data_format is used to set the value of xshape which is used in the test. Perhaps I misunderstood the question, is there a particular variable that is unused? Can you add a link to a line in the codebase?

**Comment by mikesol** [comment]
Hey!
Thank you very much for your reply.
The test being run on different shapes makes total sense, but I do not understand why the `data_format` language is there.  Meaning that you could rewrite the test with the language below to get rid of `data_format`. Unless `data_format` is supposed to communicate some useful information about a Keras spec or expected usage of batch normalization, but if this is the case, then there should be a comment about that.
```
        for x_shape in [(1, 4, 2, 3), (1, 2, 3, 4)]:
            x_val = np.random.random(x_shape).astype(np.float32)
            xth = KTH.variable(x_val) if KTH else None
            xtf = KTF.variable(x_val) if KTF else None
            xc = KC.placeholder(x_shape) if KC else None
            xm = KM.placeholder(x_shape) if KM else None
            zth, _, _ = (None,)*3 if KTH is None else KTH.normalize_batch_in_training(xth, None, None,
                                                        reduction_axes='per-activation')
            ztf, _, _ = (None,)*3 if KTF is None else KTF.normalize_batch_in_training(xtf, None, None,
                                                        reduction_axes=[0, 1, 2, 3])
            zc, _, _ = (None,)*3 if KC is None else KC.normalize_batch_in_training(xc, None, None,
                                                      reduction_axes=[0, 1, 2, 3])
            zm, _, _ = (None,)*3 if KM is None else KM.normalize_batch_in_training(xm, None, None,
                                                      reduction_axes=[0, 1, 2, 3])
            
            zth = None if KTH is None else KTH.eval(zth)
            ztf = None if KTF is None else KTF.eval(ztf)
            zc = None if KC is None else KC.function([xc], [zc])([x_val])[0]
            zm = None if KM is None else KM.eval(zm)
            res = [x for x in (zth, ztf, zc, zm) if x is not None]
            for x in range(len(res[:-1])):
                for y in range(x+1, len(res)):
                    assert res[x].shape == res[y].shape
```

**Comment by fchollet** [comment]
This is a good point, `data_format` should affect the choice of `reduction_axes` value.

**Comment by fchollet** [comment]
Essentially:

```python
if data_format == 'channels_last':
    reduction_axes = [0, 1, 2]
else:
    reduction_axes = [0, 2, 3]
```

**Comment by mikesol** [comment]
Cool!
I'll make a pull request next week.
Btw and somewhat off-topic...
I wrote a short article about Meeshkan's experience in Keras backend writing that is on https://hackernoon.com/the-meeshkan-keras-backend-covers-100-of-the-keras-public-api-why-this-matters-8207cb601f19. There are several things in that article that I'd be happy to help integrate into Keras, but I'm not sure where you are in your roadmap or how valid you feel the points I make are.  If you feel any of that could be useful, please let me know what bits are pertinent and I can formalize them into a series of pull requests (it's mostly stuff about documentation, though some parts would require minor refactoring of the backend code).
